impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -307,30 +336,101 @@ public Client build() {
             this.cacheManager = cacheManagerBuilder.build();
         }
 
-        if (this.clientConfig.getClientCredentialsResolver() == null && this.clientCredentials != null) {
+        if (isOAuth2Flow()) {
+            this.clientConfig.setClientCredentialsResolver(new OAuth2ClientCredentialsResolver(this.clientCredentials));
+        }
+        else if (this.clientConfig.getClientCredentialsResolver() == null && this.clientCredentials != null) {
             this.clientConfig.setClientCredentialsResolver(new DefaultClientCredentialsResolver(this.clientCredentials));
         }
         else if (this.clientConfig.getClientCredentialsResolver() == null) {
-            this.clientConfig.setClientCredentialsResolver(new DefaultClientCredentialsResolver(clientConfig));
+            this.clientConfig.setClientCredentialsResolver(new DefaultClientCredentialsResolver(this.clientConfig));
         }
 
         if (this.clientConfig.getBaseUrlResolver() == null) {
             ConfigurationValidator.assertOrgUrl(this.clientConfig.getBaseUrl(), allowNonHttpsForTesting);
             this.clientConfig.setBaseUrlResolver(new DefaultBaseUrlResolver(this.clientConfig.getBaseUrl()));
         }
 
+        // OAuth2
+        if (isOAuth2Flow()) {
+            this.clientConfig.setAuthenticationScheme(AuthenticationScheme.OAUTH2_PRIVATE_KEY);
+
+            validateOAuth2ClientConfig(this.clientConfig);
+
+            accessTokenRetrieverService = new AccessTokenRetrieverServiceImpl(clientConfig);
+
+            OAuth2ClientCredentials oAuth2ClientCredentials =
+                new OAuth2ClientCredentials(accessTokenRetrieverService);
+            OAuth2ClientCredentialsResolver oAuth2ClientCredentialsResolver =
+                new OAuth2ClientCredentialsResolver(oAuth2ClientCredentials);
+            this.clientConfig.setClientCredentialsResolver(oAuth2ClientCredentialsResolver);
+        }
+
         return new DefaultClient(clientConfig, cacheManager);
     }
 
+    /**
+     * @since 0.6.0
```

Comment:
```suggestion
     * @since 1.6.0
```
I wouldn't worry too much about adding the `@since` tags to private methods, it's mostly for the public API